Schemas.md

```diff
@@ -0,0 +1,521 @@
+       'DELETE': {
+           'username': {
+               'schema': {
+                   'type': 'basestring'
+               }
+           }
+       }
+   }
+   ```
+
+#### Important code pointers
+
+When adding schemas for the args of a particular handler class, some analysis is typically needed. The following points discuss the conventions adopted throughout the codebase for adding schemas to handler classes. **Please read these conventions carefully**.
+
+##### Default and optional arguments
+
+If an argument is not present in a payload/request, and the schema for that argument is defined in the handler, then that argument is treated as “missing”. For missing args, schema utils will raise AssertionError which is represented as InvalidInputException by vaidate_and_normalize_args() method.
+
+To provide default args for a handler, include a key with the name ```default_value``` along with the schema. The value for this key is the default value with which the arg will be updated if no value for that arg is provided in the request. If an argument is optional and it is not supposed to be updated with any default value, then the “default_value” key should contain None.
+
+**Example when default value is provided**: Let ```apply_draft``` be an optional argument which should take the default value False if no value for that arg is provided in the request/payload. In that case, the schema for "apply_draft" should look like:
+
+```python
+{
+    'GET': {
+        'apply_draft': {
+            'schema': {
+                'type': 'bool'
+             },
+            'default_value': False
+        }
+    }
+}
+```
+
+**Example when default value is not provided**: Suppose ```make_community_owned``` is an optional argument which should not take any default value if no value for that arg is provided in the request/payload. In that case, the schema for "make_community_owned" should look like:
+
+```python
+{
+    'PUT':{
+        'make_community_owned': {
+            'schema': {
+                'type': 'bool'
+             },
+            'default_value': None
+        }
+    }
+}
+```
+
+For more understanding [refer to examples](#examples-for-reference).
+
+##### Domain object arguments: Schemas with type object_dict
+
+Objects which are represented by classes written in the domain layer of the codebase are called domain objects. These classes typically include methods to validate their objects.
+
+For validating domain objects through SVS architecture, there are two preferred solutions, each for their unique cases.
+
+###### Case 1: object_class key
+
+The data coming from the payloads/requests is in the dict format and many of the domain objects do not get initialized with the dictionary form of the data so they must be initialized by using the from_dict() method of the same domain class. In this case class is directly passed into the schema with a schema key named ‘object_class’. Schema for these cases should have the two keys as follows:
```

Comment:
First sentence is very run-on, and there seem to be a bunch of missing "the"s. It might be worth doing a grammar audit on this and the rest of this doc?

Reply:
done